jina/types/arrays/neural_ops.py

```diff
@@ -87,3 +87,63 @@ def match(
                     d.pop('matches')
                 d.scores[m_name] = _dist
                 _q.matches.append(d)
+
+    def match_fast(
+        self,
+        darray: Union['DocumentArray', 'DocumentArrayMemmap'],
+        metric: Union[
+            str, Callable[['np.ndarray', 'np.ndarray'], 'np.ndarray']
+        ] = 'cosine',
+        limit: Optional[int] = inf,
+        normalization: Optional[Tuple[int, int]] = None,
+        use_scipy: bool = False,
+        metric_name: Optional[str] = None,
+    ) -> Union['DocumentArray', 'DocumentArrayMemmap']:
+        """
+        Alternative match that returns does not append matches
```

Comment:
?

Reply:
This is not meant to be an update for match, just some code to show the bottlenecks